lib/galaxy/visualization/plugins/config_parser.py

```diff
@@ -0,0 +1,462 @@
+import galaxy.model
+from galaxy import util
+
+import logging
+log = logging.getLogger( __name__ )
+
+
+class ParsingException( ValueError ):
+    """
+    An exception class for errors that occur during parsing of the visualizations
+    framework configuration XML file.
+    """
+    pass
+
+
+class VisualizationsConfigParser( object ):
+    """
+    Class that parses a visualizations configuration XML file.
+
+    Each visualization will get the following info:
+        - how to load a visualization:
+            -- how to find the proper template
+            -- how to convert query string into DB models
+        - when/how to generate a link to the visualization
+            -- what provides the data
+            -- what information needs to be added to the query string
+    """
+    #: show/suppress info/warnings about deprecated elements and settings
+    show_deprecations = False
```

Comment:
Is this configurable anywhere?  I'm not seeing it.  And, since it's only used the one time, I'd just leave it up to loglevel to show (or not) -- that is, maybe use log.debug instead of log.info (as below) if we want it to be mostly quiet.